Clarify --dir requirements in README.md

chrisranderson opened the pull request:
Since Ventura is pretty new, I think it'd be good to warn people before they download a large image. `tart run --help` gives me this:

```
  --dir <name:path[:ro]>  Additional directory shares with an optional read-only specifier
                          (e.g. --dir="build:~/src/build" --dir="sources:~/src/sources:ro")
        Requires host to be macOS 13.0 (Ventura) or newer.
        All shared directories are automatically mounted to "/Volumes/My Shared Files" directory on macOS,
        while on Linux you have to do it manually: "mount -t virtiofs com.apple.virtio-fs.automount /mount/point".
        For macOS guests, they must be running macOS 13.0 (Ventura) or newer.
```

I'm unclear if `--dir` requires both host and guest to be running Ventura or if the "For macOS guests..." is only talking about the "All shared directories..." portion.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/cirruslabs/tart?pullRequest=376) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/cirruslabs/tart?pullRequest=376) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/cirruslabs/tart?pullRequest=376) it.</sub>

chrisranderson commented:
If/when this is approved, I'll request CLA approval from my legal peeps.

edigaryev commented:
I think this is already mentioned in the `README.md` just a couple lines below:

>Note: to use the directory mounting feature, the host needs to run macOS 13.0 (Ventura) or newer.

chrisranderson commented:
It's a nitpick, but I think it's too low - you only need to read this portion to know what to do:

```
To mount a directory, run the VM with the --dir argument:

tart run --dir=project:~/src/project vm
```

The warning should come before usage, but it's currently several paragraphs below the example.

chrisranderson commented:
Is there a way to accomplish the same thing on < macOS 13.0 using `.cirrus.yml` and `cirrus run`? Documenting an alternative would be great.

edigaryev commented:
There's also a requirement for the guest to be macOS 13.0 in the "Accessing mounted directories in macOS guests" section:

>Note: to use the directory mounting feature, the guest VM needs to run macOS 13.0 (Ventura) or newer.

If we're removing the old note about host requirements and moving it upper in shorter form, I think the note about guest requirements needs to be dealt with too.

chrisranderson commented:
Fixed the note about guest requirements, adjusted it to match your style more closely, and put the OS at the beginning of the sentence to catch eyes.

I think it's okay only to put the warning there and remove the later note since you need to do that first before accessing the directory from the guest.

edigaryev commented:
Thanks! It looks nicer now.

However, the statement doesn't seem to be valid now for users that run Linux guests: they don't need macOS 13 (Ventura) guest to be able to mount directories.

Any ideas how this can be re-phrased to fix that?
